### PR TITLE
First cut at runfiles support in pkg_* rules

### PR DESCRIPTION
**Pull request by aiuto**
Add include_runfiles support to the manifest builder used by pkg_tar and pkg_zip.  This pulls in runfiles in the same layout as bazel does. This is not a great layout, but it is understandable.  For example, an executable with a `data` dep might look like
```
foo/an_exectuable
foo/an_executable.runfiles/foo/an_excutable
foo/an_executable.runfiles/foo/data.txt
```
Note the duplication of an_executable.  That could be expensive, so we will eventually want a way to suppress that.

Next steps:
- Add e2e tests for `pkg_tar` and `pkg_zip`.
- Add `include_runfiles` to `pkg_files`. 

See #153  for overall design and features to be implemented.

### CODE REVIEW

**Comment by depthwise** [comment]
So I've tried this with the most recent rules_python to package a Python binary runfiles, and here's the problem I'm encountering: there's `DefaultInfo.default_runfiles.root_symlinks` that rules_pkg doesn't seem take into account. This breaks `py_binary` in 2 ways:
  1. `bazel_tools/tools/python/py3wrapper.sh` is missing (although it is present under external, just not directly at root)
  2. Python bootstrap expects the various external PIP modules to be in runfiles root (they are symlinked when building in bazel), but because these are, well, `root_symlinks`, they're missing too.

So here's a suggestion: perhaps this should take `root_symlinks` into account? I know this property is undocumented, but it is very much a thing.

**Comment by aiuto** [comment]
That's going to have to be a second pass.
When we start doing symlinks instead of files, it's not obvious what is
correct:
- create symlink and hope the file is there
- copy the file to the expected place and create the link
- copy the target to the location of the symlink and make it a real file.

The first choice is probably wrong.
If the second choice, what if the expected place is outside the workspace?
The third one seems right, and yet still wrong.

It's going to take some poking about to cover all the issues.

File: pkg/private/pkg_files.bzl [hunk 1]
```diff
@@ -259,6 +259,10 @@ def add_label_list(
       default_group: fallback mode to use for Package*Info elements without group
     """
 
+    if hasattr(ctx.attr, "include_runfiles"):
+        include_runfiles = ctx.attr.include_runfiles
+    else:
+        include_runfiles = False
```
Comment (nacl) [inline pkg/private/pkg_files.bzl:265]: This could be shortened to

```starlark
include_runfiles = getattr(ctx, "include_runfiles", False)
```

**Review by nacl — APPROVED** [review]
LGTM all things considered, but I think the commit message heading could be made more specific and call out exactly what happened.

BTW, it looks like `pkg_zip` doesn't even support `include_runfiles` as an attribute, so that may need to be added in and tested.

File: pkg/private/pkg_files.bzl [hunk 2]
```diff
@@ -332,14 +339,29 @@ def add_from_default_info(
...
-                f,
+                dest_path = d_path,
+                src = f,
                 origin = src.label,
                 mode = fmode,
                 user = default_user,
                 group = default_group,
             )
-
+    if include_runfiles:
```
[the remark this answers is not in the document]
Comment (aiuto) [reply pkg/private/pkg_files.bzl:349]: I want to deal with factoring it out later.
I expect that I don't know the exact needs yet.
I am wondering if pkg_files should use add_label_list and pass in a lambda which can do the remapping.

File: pkg/private/pkg_files.bzl (hunk 1 above)
Comment (aiuto) [reply pkg/private/pkg_files.bzl:268]: I changed my mind.   That change actually gives a different result. The getattr with keeps yielding False, even when set.

**Pull request merged**